### PR TITLE
making extract_verbs.sh file common to any language

### PR DESCRIPTION
**Pull request by kkasunperera**

### CODE REVIEW

File: extract_verbs.sh [hunk 1]
```diff
@@ -3,10 +3,14 @@
 set -e
 
 cd ..
-# Download latest Italian Wikipedia dump
-wget http://download.wikimedia.org/itwiki/latest/itwiki-latest-pages-articles.xml.bz2
+# Download latest Wikipedia dump in language of Chioce
+echo -n "Enter wikipedia dump URL e.g.  http://download.wikimedia.org/itwiki/latest/itwiki-latest-pages-articles.xml.bz2> "
```
Comment (marfox) [inline extract_verbs.sh]: This should be replaced by a command line argument.
The script should not be interactive.

File: extract_verbs.sh [hunk 2]
```diff
@@ -14,7 +18,9 @@ find extracted -type f -exec cat {} \; > all-extracted.txt
 # Extract verbs with TreeTagger
 # N.B. treetagger segfaults with the single big file, run it over each article instead
 #cat all-extracted.txt | treetagger/cmd/tree-tagger-italian | grep VER | sort -u > verbi.txt
-find extracted -type f -exec bash -c "cat '{}' | treetagger/cmd/tree-tagger-italian | grep VER >> verbi.txt" \;
+echo -n "enter corresponding language tree-tagger path e.g. /home/kasun/tree_tagger/cmd/tree-tagger-italian "
```
Comment (marfox) [inline extract_verbs.sh]: The TreeTagger language should be inferred from the used Wikipedia dump.

File: extract_verbs.sh [hunk 3]
```diff
@@ -14,9 +71,9 @@ find extracted -type f -exec cat {} \; > all-extracted.txt
...
 # N.B. treetagger segfaults with the single big file, run it over each article instead
 #cat all-extracted.txt | treetagger/cmd/tree-tagger-italian | grep VER | sort -u > verbi.txt
-find extracted -type f -exec bash -c "cat '{}' | treetagger/cmd/tree-tagger-italian | grep VER >> verbi.txt" \;
+find extracted -type f -exec bash -c "cat '{}' | treetagger/cmd/$TAGGER | grep VER >> verbi.txt" \;
 sort -u verbi.txt > unique-sorted-verbs.txt
 # Extract vocabulary
 python scripts/bag_of_words.py all-extracted.txt
 # POS tagging + chunker with TextPro
-perl textpro.pl -verbose -html -l ita -c token+sentence+pos+chunk -o . ~/srl/training/itwiki/gold
+perl textpro.pl -verbose -html -l ita -c token+sentence+pos+chunk -o . ~/srl/training/"$LANGSHORTCODE"wiki/gold
```
Comment (kkasunperera) [inline extract_verbs.sh]: @marfox  I have modified the script to your updated task pls check.

File: extract_verbs.sh [hunk 4]
```diff
@@ -3,10 +3,67 @@
...
 cd ..
-# Download latest Italian Wikipedia dump
-wget http://download.wikimedia.org/itwiki/latest/itwiki-latest-pages-articles.xml.bz2
+# Download latest Wikipedia dump in language of Chioce
+echo "Enter wikipedia dump language e.g. en OR english"
+read LANGCODE
+#convert lnaguge code to lower case
+LANGCODE="$(echo $LANGCODE | tr '[A-Z]' '[a-z]')"
+
+#looping to select language
```
Comment (marfox) [inline extract_verbs.sh]: Please use a case statement here, I think it make the code cleaner.

File: extract_verbs.sh [hunk 5]
```diff
@@ -3,10 +3,67 @@
 set -e
 
 cd ..
-# Download latest Italian Wikipedia dump
-wget http://download.wikimedia.org/itwiki/latest/itwiki-latest-pages-articles.xml.bz2
+# Download latest Wikipedia dump in language of Chioce
```
Comment (marfox) [inline extract_verbs.sh]: typo here, should be "language of **choice**"

File: extract_verbs.sh [hunk 6]
```diff
@@ -3,10 +3,67 @@
...
+	LANGSHORTCODE="ru"
+	TAGGER="tree-tagger-russian"
+elif [ "$LANGCODE" = "slovak" ] || [ "$LANGCODE" = "sk" ]; then
+	LANGSHORTCODE="sk"
+	TAGGER="tree-tagger-slovak"
+elif [ "$LANGCODE" = "spanish" ] || [ "$LANGCODE" = "es" ]; then
+	LANGSHORTCODE="es"
+	TAGGER="tree-tagger-spanish"
+else
+	echo "Enterned Languge is not yet supported in this script! EXIT from script"
```
Comment (marfox) [inline extract_verbs.sh]: typo here, should be "**Entered language**"

File: extract_verbs.sh (hunk 4 above)
Comment (kkasunperera) [inline extract_verbs.sh]: @marfox they reason why I didn't put a case statement is user can enter long name of the language or short code. For e.g. english or en. Making this is a switch-case statement makes code more lengthy (onle language two cases). This way it is much compact, do you still want to modify it as a switch-case ?

Comment (marfox) [inline extract_verbs.sh]: I think you can do it like this:
```bash
case "$LANGCODE" in
    "en" | "english" )
        ...
esac
```
See example 4 [here](http://www.thegeekstuff.com/2010/07/bash-case-statement/).
Please test it and go ahead.

File: extract_verbs.sh (hunk 3 above)
Comment (kkasunperera) [inline extract_verbs.sh]: @marfox made the suggested case statement change.

File: extract_verbs.sh [hunk 7]
```diff
@@ -3,10 +3,97 @@
 set -e
 
 cd ..
-# Download latest Italian Wikipedia dump
-wget http://download.wikimedia.org/itwiki/latest/itwiki-latest-pages-articles.xml.bz2
+# Download latest Wikipedia dump in language of choice
+echo "Enter wikipedia dump language e.g. en OR english"
```
Comment (marfox) [inline extract_verbs.sh]: This should be replaced by a command line argument.
The script should not be interactive.

File: extract_verbs.sh [hunk 8]
```diff
@@ -1,12 +1,106 @@
...
-wget http://download.wikimedia.org/itwiki/latest/itwiki-latest-pages-articles.xml.bz2
+# Download latest Wikipedia dump in language of choice
+
+if [ "$#" -ne 1 ]; then
+	echo "Usage: sh extract_verbs.sh <language code>"
+	exit
+fi
+
+LANGCODE=$1
+
```
Comment (kkasunperera) [inline extract_verbs.sh]: @marfox this should be fine?

**Pull request merged**